packages/contracts/contracts/L1/interfaces/IAssetManager.sol

```diff
@@ -9,101 +9,72 @@ interface IAssetManager {
-        uint256[] claimRequestTimes;
-        mapping(uint256 => uint128) pendingValidatorRewards;
+        uint128 rewardPerKghStored;
     }
 
     /**
      * @notice Constructs the delegator of KRO in the vault of a validator.
      *
-     * @custom:field shares                 Amount of shares for KRO delegation.
-     * @custom:field undelegateRequestTimes Timestamps of undelegation requests.
-     * @custom:field pendingKroShares       A mapping of timestamp undelegations are requested to
-     *                                      pending KRO shares for undelegation.
+     * @custom:field shares          Amount of shares for KRO delegation.
+     * @custom:field lastDelegatedAt Last timestamp when the delegator delegated. The delegator can
+     *                               undelegate after UNDELEGATION_DELAY_SECONDS elapsed.
      */
     struct KroDelegator {
         uint128 shares;
-        uint256[] undelegateRequestTimes;
-        mapping(uint256 => uint128) pendingKroShares;
-    }
-
-    /**
-     * @notice Represents the amounts of KRO and KGH shares for KGH delegator.
-     *
-     * @custom:field kro Amount of KRO shares.
-     * @custom:field kgh Amount of KGH shares.
-     */
-    struct KghDelegatorShares {
-        uint128 kro;
-        uint128 kgh;
+        uint128 lastDelegatedAt;
     }
 
     /**
      * @notice Constructs the delegator of KGH in the vault of a validator.
      *
-     * @custom:field shares                 Amount of shares for KRO in KGH and KGH itself to the
-     *                                      corresponding tokenId.
-     * @custom:field undelegateRequestTimes Timestamps of undelegation requests.
-     * @custom:field pendingKghIds          A mapping of timestamp undelegations are requested to
-     *                                      token ids of KGHs for undelegation.
-     * @custom:field pendingShares          A mapping of timestamp undelegations are requested to
-     *                                      pending KRO and KGH shares for undelegation.
+     * @custom:field lastDelegatedAt   Last timestamp when the delegator delegated. The delegator
+     *                                 can undelegate after UNDELEGATION_DELAY_SECONDS elapsed.
+     * @custom:field rewardPerKghPaid  Accumulated paid boosted reward per 1 KGH.
+     * @custom:field kghNum            Total number of KGH delegated.
+     * @custom:field delegationHistory A mapping of tokenId to the delegation timestamp.
+     * @custom:field kroShares         A mapping of tokenId to the amount of shares for KRO in KGH.
      */
     struct KghDelegator {
-        mapping(uint256 => KghDelegatorShares) shares;
-        uint256[] undelegateRequestTimes;
-        mapping(uint256 => uint256[]) pendingKghIds;
-        mapping(uint256 => KghDelegatorShares) pendingShares;
+        uint128 lastDelegatedAt;
+        uint128 rewardPerKghPaid;
+        uint256 kghNum;
+        mapping(uint256 => uint128) delegationHistory;
```

Comment:
Why do we need this?

Reply:
Since `kroShares` can be 0, there are no way to confirm if a tokenId is delegated exactly by the delegator without this mapping. Welcome for any other idea!

Reply:
If we have `delegationHistory`, we can remove `lastDelegatedAt` and manage undelegation using the history mapping